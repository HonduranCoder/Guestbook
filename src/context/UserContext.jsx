import { useContext, createContext, useState } from 'react';

// create our context
const UserContext = createContext();
// create a user provider that takes in the children
// create text
// use the user provider context
const UserProvider = ({ children }) => {
  const [user, setUser] = useState('');
  return (
    <UserContext.Provider value={{ user, setUser }}>
      {children}
    </UserContext.Provider>
  );
};

// custom hook
const useUser = () => {
  const context = useContext(UserContext);

  if (context === undefined) {
    throw new Error('Provider needed');
  }
  return context;
};

export { UserProvider, useUser };
