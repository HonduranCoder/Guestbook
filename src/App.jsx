import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import './App.css';
import PrivateRoute from './components/PrivateRoute/PrivateRoute.jsx';
import { ProvideAuth } from './context/ProvideAuth';
import Home from './views/Home/Home';
import Layout from './views/Layout/Layout';
import Auth from './views/Auth/Auth.jsx';

function App() {
  return (
    <ProvideAuth>
      <Router>
        <Layout>
          <Switch>
            <Route path="/login">
              <Auth />
            </Route>
            <PrivateRoute path="/">
              <Home />
            </PrivateRoute>
          </Switch>
        </Layout>
      </Router>
    </ProvideAuth>
  );
}

export default App;
