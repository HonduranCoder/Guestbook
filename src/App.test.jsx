import React from 'react';
import { render } from '@testing-library/react';
import { UserProvider } from './context/UserContext.jsx';
import App from './App.jsx';

it('should render App', () => {
  const { container } = render(
    <React.StrictMode>
      <UserProvider>
        <App />
      </UserProvider>
    </React.StrictMode>
  );
  expect(container).toMatchSnapshot();
});
