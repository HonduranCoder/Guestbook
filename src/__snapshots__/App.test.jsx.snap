// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render App 1`] = `
<div>
  <div>
    <header>
      <p>
        Sign in
      </p>
    </header>
    <main>
      <h1>
        Please Sign the Guestbook!
      </h1>
      <form>
        <div>
          <div>
            <label>
              Guest Name
            </label>
          </div>
          <div
            class="md:w-2/3"
          >
            <input
              id="guestName"
              placeholder="Guest Name"
              type="text"
              value=""
            />
          </div>
        </div>
        <label>
          Guest Entry
        </label>
        <div>
          <textarea
            id="newEntry"
            placeholder="Your Entry!"
          />
        </div>
        <div>
          <button>
            Sign
          </button>
          
        </div>
      </form>
      <div>
        <ul />
      </div>
    </main>
  </div>
</div>
`;
